flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/KubernetesOperatorSessionJobConfigOptions.java

```diff
@@ -0,0 +1,44 @@
+package org.apache.flink.kubernetes.operator.config;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/** List supported session job specific configurations */
+public enum KubernetesOperatorSessionJobConfigOptions {
+
+    SESSION_JOB_HTTP_JAR_HEADERS("kubernetes.operator.user.artifacts.http.header");
```

Comment:
I think this all can be simply part of the `KuberenetesOperatorConfigOptions` no need for a new class

Reply:
Hi @gyfora, I was thinking if we put this configuration at the `KuberenetesOperatorConfigOptions`, Does that imply you intend to put this header config field as a candidate in `flink-conf.yaml`? In that case, we are assuming the headers are the same for all session jobs within the cluster? I think maybe we can allow users to config different headers for different session jobs within one session cluster.

Reply:
Hi @gyfora If not, you also intend to keep this new `flinkConfiguration` as job specific. How do we control the behavior in the session cluster since it could have multiple different jobs?
Taking a step back, we can assume the headers are all the same for the same cluster, that will make things simplier, but might NOT be very flexible.
cc @wangyang0918 
See Yang's comment.
https://issues.apache.org/jira/browse/FLINK-27483?focusedCommentId=17532137&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17532137

Reply:
I am not suggesting to change the logic, just the location of the constant you defined to not create unnecessary new classes and spread the options too much :)

The operator should always use the config defined in the resource (session job/deployment) no difference. We usually apply this logic on top of the default config defined in the operator.

Reply:
But I would allow users to put the header conf in the default Flink conf yaml to have some default value applied in case the session job doesn’t want to ovverride it

Reply:
We just need to document clearly which configuration options could take effect in `FlinkSessionJob.spec. flinkConfiguration`.